src/main/java/com/edmunds/tools/databricks/maven/model/JobTemplateModel.java

```diff
@@ -36,22 +36,34 @@
     private String groupId;
     private String artifactId;
     private String version;
+    // TODO This property should not be persisted with the template model
     private String environment;
     private String groupWithoutCompany;
+    // The rationale for persisting these properties is because a deployed artifact will already have been deployed
+    // to a specific place. You cannot change that after the fact!
+    private String databricksRepo;
+    private String databricksRepoKey;
 
     /**
      * Don't use this - it's for jackson deserialization only!
      */
     public JobTemplateModel() {
     }
 
-    public JobTemplateModel(MavenProject project) {
+    public JobTemplateModel(MavenProject project,
+                            String environment, String databricksRepo, String databricksRepoKey) {
         this.groupId = project.getGroupId();
         this.artifactId = project.getArtifactId();
         this.projectProperties = project.getProperties();
         this.systemProperties = System.getProperties();
         this.version = defaultString(systemProperties.getProperty(DEPLOY_VERSION), project.getVersion());
         this.groupWithoutCompany = stripCompanyPackage(project.getGroupId());
+        this.databricksRepo = databricksRepo;
+        this.databricksRepoKey = databricksRepoKey;
+        this.environment = environment;
+        //TODO NEED TO GET RID OF this once we are ready. This is for backwards compatibility
+        projectProperties.setProperty("databricks.repo", databricksRepo);
```

Comment:
This is ONLY necessary for job settings files. If job settings files were changed to reference "databricksRepo" directly, then it would work as is.

I could change the default job settings to demonstrate that.